web/react/components/create_post.jsx

```diff
@@ -112,6 +131,63 @@ module.exports = React.createClass({
     handleUserInput: function(messageText) {
         this.resizePostHolder();
         this.setState({messageText: messageText});
+
+        // look to see if the message begins with any carets to indicate that it's a reply
+        var replyMatch = messageText.match(/^\^+/g);
+        if (replyMatch) {
+            // the number of carets indicates how many message threads back we're replying to
+            var caretCount = replyMatch[0].length;
+
+            var posts = PostStore.getCurrentPosts();
+
+            var rootId = "";
+
+            // find the nth most recent post that isn't a comment on another (ie it has no parent) where n is caretCount
+            for (var i = 0; i < posts.order.length; i++) {
+                var postId = posts.order[i];
+
+                if (posts.posts[postId].parent_id === "") {
+                    if (caretCount == 1) {
+                        rootId = postId;
+                        break;
+                    } else {
+                        caretCount -= 1;
+                    }
+                }
+            }
+
+            if (rootId) {
+                // only dispatch an event if something changed
+                if (rootId != this.state.rootId) {
+                    // set the parent id to match the root id so that we're replying to the first post in the thread
+                    var parentId = rootId;
```

Comment:
I'm not sure exactly. When you reply through the sidebar, it also sets them to be the same. I'll try to figure out why that is.